Support for importing common js modules.

unageanu opened the pull request:
How to use common js module distributed in npm :
1.  add dependencies in a `package.json` .
   
   ```
   "dependencies": {
      "foo": "^1.0.0"
   }
   ```
2. add dependent files to commonJsModuleList in `src/source-list.js` .
   
   ```
   var commonJsModuleList = [
       "node_modules/foo/lib/foo.js"
   ];
   ```
3. add dependent files to a `scripts/build` entry in the package.json.
   
   ```
   "scripts": {
       build:  "mkdirp lib && java ... node_modules/foo/lib/foo.js",
       ...
   ```
4. import and use the dependent library using `require()` .
   
   ```
   var foo = require("node_modules/foo/lib/foo");
   ```

unageanu committed:
Upgrade Closure Compiler.

- for fix [this issue](https://github.com/google/closure-compiler/pull/1287).

unageanu committed:
Upgrade Closure Compiler.

- for fix [this issue](https://github.com/google/closure-compiler/pull/1287).

unageanu committed:
Support Common JS modules.

unageanu committed:
Support Common JS modules.

unageanu committed:
Suppress compiler warnings.

unageanu committed:
Suppress compiler warnings.

unageanu committed:
Suppress eslint errors.

unageanu committed:
Suppress eslint errors.

kwkbtr committed:
Fix bug on page navigation when the document contains cross references

kwkbtr committed:
Merge pull request #306 from vivliostyle/fix_page_navigation_error

Fix bug on page navigation when the document contains cross references

kwkbtr committed:
Fix viewport zoom behavior when viewport size is explicitly specified

kwkbtr committed:
Update Chromium version

kwkbtr committed:
Update version

kwkbtr committed:
Update Change Log

kwkbtr committed:
Update version

kwkbtr committed:
Fix bug that user/author stylesheets are not used when a Document object is provided

Additional user/author stylesheets should be set before the document is loaded.

unageanu committed:
Support CommonJS modules on Karma by using karma-commonjs.

unageanu committed:
Support CommonJS modules on Karma by using karma-commonjs.

kwkbtr reviewed (changes requested):
Please rebase to (or merge) master and force push since this branch currently conflicts with master.

unageanu committed:
Remove unused type annotation.

unageanu committed:
Remove unused type annotation.

unageanu committed:
Remove karma-commonjs settings for avoiding error.

- If you require commonjs modules, Add settings below:
  - Add `commonjs` to `frameworks`.
    ```
    frameworks: ["jasmine", 'commonjs'],
    ```
  - Add `preprocessors` and `commonjsPreprocessor` setting.
    ```
    preprocessors: {
        "node_modules/dummy/*.js": ['commonjs']
    },
    commonjsPreprocessor: {
        modulesRoot: './'
    },
    ```

unageanu committed:
Remove karma-commonjs settings for avoiding error.

- If you require commonjs modules, Add settings below:
  - Add `commonjs` to `frameworks`.
    ```
    frameworks: ["jasmine", 'commonjs'],
    ```
  - Add `preprocessors` and `commonjsPreprocessor` setting.
    ```
    preprocessors: {
        "node_modules/dummy/*.js": ['commonjs']
    },
    commonjsPreprocessor: {
        modulesRoot: './'
    },
    ```

unageanu committed:
Merge branch 'support_common_js_modules' of https://github.com/vivliostyle/vivliostyle.js into support_common_js_modules

Conflicts:
	src/adapt/ops.js

kwkbtr committed:
Suppress warnings of Closure Compiler and ESLint
